### PR TITLE
Adding installation script tests

### PR DESCRIPTION
**Pull request by ghost**
When we materially change the way the installation scripts install the SDK for 3.0 and beyond - these tests, and the ability to add more installation script tests, will assist with development.

### CODE REVIEW

File: test/InstallationScript.Tests/InstallationScript.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <TargetFramework>$(CliTargetFramework)</TargetFramework>
+    <AssemblyName>InstallationScript.Tests</AssemblyName>
```
Comment (livarcocc) [inline test/InstallationScript.Tests/InstallationScript.Tests.csproj]: I don't think you need AssemblyName and AssetTargetFallBack anymore.

File: test/InstallationScript.Tests/InstallationScript.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    <AssetTargetFallback>$(AssetTargetFallback);dotnet5.4;portable-net451+win8</AssetTargetFallback>
+  </PropertyGroup>
+
+
+  <ItemGroup>
+    <ProjectReference Include="..\Microsoft.DotNet.Tools.Tests.Utilities\Microsoft.DotNet.Tools.Tests.Utilities.csproj" />
```
Comment (livarcocc) [inline test/InstallationScript.Tests/InstallationScript.Tests.csproj]: Are all these references really needed?

File: Microsoft.DotNet.Cli.sln [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.28515.72
+# Visual Studio 15
```
Comment (livarcocc) [inline Microsoft.DotNet.Cli.sln]: Given that VS 16 is out the door, should we stick it to it? Not really keen on having this file continuously flipped depending on who is editing it.

File: test/InstallationScript.Tests/GivenThatIWantToInstallTheSdkFromAScript.cs [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+            {
+                args.Add(value);
+            }
+
+            var commandResult = CreateInstallCommand(args)
+                            .CaptureStdOut()
+                            .CaptureStdErr()
+                            .Execute();
+
+            Console.WriteLine(commandResult.StdOut);
```
Comment (livarcocc) [inline test/InstallationScript.Tests/GivenThatIWantToInstallTheSdkFromAScript.cs]: Do you really need to output this to the console?

File: test/InstallationScript.Tests/InstallationScript.Tests.csproj (hunk 2 above)
Comment (johnbeisner) [reply test/InstallationScript.Tests/InstallationScript.Tests.csproj]: These were from experimentations, forgot to remove them - thanks.

File: Microsoft.DotNet.Cli.sln (hunk 3 above)
Comment (johnbeisner) [reply Microsoft.DotNet.Cli.sln]: Done

File: test/InstallationScript.Tests/GivenThatIWantToInstallTheSdkFromAScript.cs [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+        public void WhenRuntimeParametersArePassedToInstallScripts(string runtimeParameters, string resolvedRuntimeType)
+        {
+            var args = new List<string> { "-dryrun", runtimeParameters };
+
+            var commandResult = CreateInstallCommand(args)
+                            .CaptureStdOut()
+                            .CaptureStdErr()
+                            .Execute();
+
+            Console.WriteLine(commandResult.StdOut);
```
Comment (livarcocc) [inline test/InstallationScript.Tests/GivenThatIWantToInstallTheSdkFromAScript.cs]: Same here about printing to console.

Comment (livarcocc) [reply test/InstallationScript.Tests/GivenThatIWantToInstallTheSdkFromAScript.cs]: Well, this is in all tests. Consider if that's really needed.

**Comment by johnbeisner** [comment]
@dotnet/dotnet-cli 
Can I get an approval to merge this PR

**Review by peterhuene — APPROVED** [review]

**Pull request merged**